test/transcoding/relationals_double.ll

```diff
@@ -3,68 +3,279 @@
+; RUN: llvm-spirv -r %t.spv -o %t.rev.spv.bc --spirv-target-env=SPV-IR
+; RUN: llvm-dis < %t.rev.spv.bc | FileCheck %s --check-prefix=CHECK-SPV-IR
 ; RUN: llvm-spirv -r %t.spv -o %t.rev.bc
 ; RUN: llvm-dis < %t.rev.bc | FileCheck %s --check-prefix=CHECK-LLVM
 
-; CHECK-SPIRV: 4 TypeInt [[IntTypeID:[0-9]+]] 64
-; CHECK-SPIRV: 4 TypeVector [[Int64VectorTypeID:[0-9]+]] [[IntTypeID]] 2
+; This test checks following SYCL relational builtins with double and double2
+; types:
+;   isfinite, isinf, isnan, isnormal, signbit, isequal, isnotequal, isgreater
+;   isgreaterequal, isless, islessequal, islessgreater, isordered, isunordered
 
-; CHECK-SPIRV: 6 Select [[Int64VectorTypeID]]
-; CHECK-SPIRV: 6 Select [[Int64VectorTypeID]]
-; CHECK-SPIRV: 6 Select [[Int64VectorTypeID]]
-; CHECK-SPIRV: 6 Select [[Int64VectorTypeID]]
+; CHECK-SPIRV: 2 TypeBool [[BoolTypeID:[0-9]+]]
+; CHECK-SPIRV: 4 TypeVector [[BoolVectorTypeID:[0-9]+]] [[BoolTypeID]] 2
 
+; CHECK-SPIRV: 4 IsFinite [[BoolTypeID]]
+; CHECK-SPIRV: 4 IsInf [[BoolTypeID]]
+; CHECK-SPIRV: 4 IsNan [[BoolTypeID]]
+; CHECK-SPIRV: 4 IsNormal [[BoolTypeID]]
+; CHECK-SPIRV: 4 SignBitSet [[BoolTypeID]]
+; CHECK-SPIRV: 5 FOrdEqual [[BoolTypeID]]
+; CHECK-SPIRV: 5 FUnordNotEqual [[BoolTypeID]]
+; CHECK-SPIRV: 5 FOrdGreaterThan [[BoolTypeID]]
+; CHECK-SPIRV: 5 FOrdGreaterThanEqual [[BoolTypeID]]
+; CHECK-SPIRV: 5 FOrdLessThan [[BoolTypeID]]
+; CHECK-SPIRV: 5 FOrdLessThanEqual [[BoolTypeID]]
+; CHECK-SPIRV: 5 FOrdNotEqual [[BoolTypeID]]
+; CHECK-SPIRV: 5 Ordered [[BoolTypeID]]
+; CHECK-SPIRV: 5 Unordered [[BoolTypeID]]
+
+; CHECK-SPIRV: 4 IsFinite [[BoolVectorTypeID]]
+; CHECK-SPIRV: 4 IsInf [[BoolVectorTypeID]]
+; CHECK-SPIRV: 4 IsNan [[BoolVectorTypeID]]
+; CHECK-SPIRV: 4 IsNormal [[BoolVectorTypeID]]
+; CHECK-SPIRV: 4 SignBitSet [[BoolVectorTypeID]]
+; CHECK-SPIRV: 5 FOrdEqual [[BoolVectorTypeID]]
+; CHECK-SPIRV: 5 FUnordNotEqual [[BoolVectorTypeID]]
+; CHECK-SPIRV: 5 FOrdGreaterThan [[BoolVectorTypeID]]
+; CHECK-SPIRV: 5 FOrdGreaterThanEqual [[BoolVectorTypeID]]
+; CHECK-SPIRV: 5 FOrdLessThan [[BoolVectorTypeID]]
+; CHECK-SPIRV: 5 FOrdLessThanEqual [[BoolVectorTypeID]]
+; CHECK-SPIRV: 5 FOrdNotEqual [[BoolVectorTypeID]]
+; CHECK-SPIRV: 5 Ordered [[BoolVectorTypeID]]
+; CHECK-SPIRV: 5 Unordered [[BoolVectorTypeID]]
+
+; CHECK-SPV-IR: call spir_func i32 @_Z16__spirv_IsFinited(double
+; CHECK-SPV-IR: call spir_func i32 @_Z13__spirv_IsInfd(double
+; CHECK-SPV-IR: call spir_func i32 @_Z13__spirv_IsNand(double
+; CHECK-SPV-IR: call spir_func i32 @_Z16__spirv_IsNormald(double
+; CHECK-SPV-IR: call spir_func i32 @_Z18__spirv_SignBitSetd(double
+; CHECK-SPV-IR: fcmp oeq double
+; CHECK-SPV-IR: fcmp une double
+; CHECK-SPV-IR: fcmp ogt double
+; CHECK-SPV-IR: fcmp oge double
+; CHECK-SPV-IR: fcmp olt double
+; CHECK-SPV-IR: fcmp ole double
+; CHECK-SPV-IR: fcmp one double
+; CHECK-SPV-IR: fcmp ord double
+; CHECK-SPV-IR: fcmp uno double
+
+; CHECK-SPV-IR: call spir_func <2 x i64> @_Z16__spirv_IsFiniteDv2_d(<2 x double>
```

Comment:
not related to this patch, I think we shall unify the return type in spirv-friendly IR for vector output:
; CHECK-SPV-IR: call spir_func <2 x i64> @_Z16__spirv_IsFiniteDv2_d(<2 x double>
; CHECK-SPV-IR: call spir_func <2 x i32> @_Z16__spirv_IsFiniteDv2_f(<2 x float>
; CHECK-SPV-IR: call spir_func <2 x i16> @_Z16__spirv_IsFiniteDv2_Dh(<2 x half>

@Naghasan how do you think?

Reply:
Could we always represent boolean vector as a vector of i8 in SPIRV Friendly IR? I think that such rule should be applied to all opcodes operating on boolean vector and should also be documented in https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/master/docs/SPIRVRepresentationInLLVM.rst.

Reply:
answered above, I agree with @aratajew, `i8` is probably the most portable type from a frontend perspective

Reply:
Intel GPU Backend also expects that boolean vectors will be represented as i8 vectors: https://github.com/intel/intel-graphics-compiler/blob/master/IGC/BiFModule/Headers/spirv.h#L153-L157 So it sounds reasonable to me to make it this way.

Reply:
if there is no objection, I can update this PR to change SPIRV Friendly IR return type of relational builtin to match with https://github.com/intel/llvm/blob/8c5b7017a925701ef4034056b5ed8e0fac2a0011/clang/lib/Sema/SPIRVBuiltins.td#L855

Reply:
> if there is no objection, I can update this PR to change SPIRV Friendly IR return type of relational builtin to match with https://github.com/intel/llvm/blob/8c5b7017a925701ef4034056b5ed8e0fac2a0011/clang/lib/Sema/SPIRVBuiltins.td#L855

done

Reply:
@MrSidims, @bader what do you think about current solution? Do you agree to have it this way? If so, it would be great to merge it.

Reply:
Let me revisit the PR by EO tomorrow.

Reply:
I think we should get approval from @Naghasan and resolve merge conflicts before we can merge it.

Reply:
merge conflict resolved. @Naghasan please review.